hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSUtil.java

```diff
@@ -1936,4 +1936,17 @@ public static boolean isParentEntry(final String path, final String parent) {
     return path.charAt(parent.length()) == Path.SEPARATOR_CHAR
         || parent.equals(Path.SEPARATOR);
   }
+
+  /**
+   * Calculate the transfer rate in megabytes/second.
+   * @param bytes bytes
+   * @param durationMS duration in milliseconds
+   * @return the number of megabytes/second of the transfer rate
+  */
+  public static long transferRateMBs(long bytes, long durationMS) {
+    if (durationMS == 0) {
```

Comment:
if it is <= 0, just return -1? Let's add a check for bytes as well.

Reply:
I dont feel we should handle other cases. This is a Utils method and any unexpected data should be left for the client to interpret. For some clients the negative values might even make sense.
The idea behind handling for durationMS = 0 is to take care of DivideByZero for cases when data transfer did not happen.

Reply:
can we specify our function as: "we expect both inputs to be positive. Otherwise, this function will return -1". 

Then returning -1 is a clear signal we don't know how to handle such inputs.

Reply:
updated